services/app-web/src/pages/SubmissionRevisionSummary/SubmissionRevisionSummaryV2.tsx

```diff
@@ -49,18 +49,25 @@ export const SubmissionRevisionSummaryV2 = (): React.ReactElement => {
         fetchPolicy: 'network-only',
     })
     const contract = fetchContractData?.fetchContract.contract
-    //We offset version by +1 of index, remove offset to find revision in revisions
+
+    // Offset version by +1 of index, remove offset to find target previous submission in the history list
     const revisionIndex = Number(revisionVersion) - 1
-    const name =
-        contract &&
-        contract?.packageSubmissions.length > Number(revisionVersion)
-            ? contract.packageSubmissions.reverse()[revisionIndex]
-                  .contractRevision.contractName
-            : ''
+
+    // Reverse revisions to get correct submission order
+    const packageSubmissions = contract? [...contract.packageSubmissions]
+        .filter((submission) => {
+            return submission.cause === 'CONTRACT_SUBMISSION'
+        })
+        .reverse() : []
+    const targetPreviousSubmission = packageSubmissions[revisionIndex] && packageSubmissions[revisionIndex].__typename ? packageSubmissions[revisionIndex] : undefined
+    const name = targetPreviousSubmission?.contractRevision.contractName
+
     useEffect(() => {
-        updateHeading({
+        // make sure you do not update the page heading until we are sure the name for that previous submission exists
+        if (name) {
+            updateHeading({
             customHeading: name,
-        })
```

Comment:
I saw necessarily re-renders happening here, ideally we dont try to update (or clear) the heading in the <header/> at all until we are sure we have the proper submission name